qt/python/mantidqt/mantidqt/widgets/instrumentview/test/test_instrumentview_presenter.py

```diff
@@ -20,14 +21,30 @@ def setUp(self) -> None:
         self.ws = mock.NonCallableMock()
         self.presenter = InstrumentViewPresenter(ws=self.ws, view=self.mock_view)
 
+    @mock.patch("mantidqt.widgets.instrumentview.presenter.AnalysisDataService")
     @mock.patch("mantidqt.widgets.instrumentview.presenter.InstrumentView")
-    def test_ws_name_passed_in_constructor(self, mock_view):
+    def test_ws_name_passed_in_constructor(self, mock_view, mock_ads):
+        mock_ads.retrieve.return_value = mock.NonCallableMock()
+
         ws = mock.NonCallableMock()
         presenter = InstrumentViewPresenter(ws)
 
         mock_view.assert_called_once_with(parent=mock.ANY, presenter=presenter,
                                           name=str(ws), window_flags=mock.ANY)
 
+    @mock.patch("mantidqt.widgets.instrumentview.presenter.AnalysisDataService")
+    @mock.patch("mantidqt.widgets.instrumentview.presenter.InstrumentView")
+    def test_constructor_works_with_ws_name(self, mock_view, mock_ads):
```

Comment:
Nice to add test for passing ws name as well as object - would it also make sense here to assert the workspace is locked/unlocked as well (jus tin case anyone decides to delete these lines however unlikely that may be!). Also I think @DavidFair told me `assertRaisesNothing` was not recommended for some reason?

Reply:
- Unit test assume anything run without `self.assertRaises` should not throw, so `assertRaisesNothing` is redundant
- It takes an exception then mangles it into a string losing some context
- It's a bespoke bit of code we have to maintain, whilst `UnitTest` already has this built in
- It makes readers assume anything not wrapped in `assertRaisesNothing` can throw an assertion during testing and be fine

So I'm trying to nip away new usage of this method with a long-term look to drop it (maybe this maintenance period?)

Reply:
I've removed the asserRaisesNothing and added a separate test for making sure the workspace aquires a readlock.